src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -330,7 +336,9 @@ private function getMutatorMethod($class, $property)
         $ucProperty = ucfirst($property);
         $ucSingulars = (array) Inflector::singularize($ucProperty);
 
-        foreach ($this->mutatorPrefixes as $prefix) {
+        $mutatorPrefixes = \in_array($ucProperty, $ucSingulars, true) ? $this->arrayMutatorPrefixesLast : $this->arrayMutatorPrefixesFirst;
```

Comment:
If the the property singulars contains the property itself, we can assume the property is singular isn'it?